src/adjoint.jl

```diff
@@ -1,23 +1,43 @@
 using MacroTools
 using MacroTools: @q, combinedef
+using ChainRulesCore: AbstractZero, Zero, DoesNotExist
 
 named(arg) = isexpr(arg, :(::)) && length(arg.args) == 1 ? :($(gensym())::$(arg.args[1])) : arg
 
 typeless(x) = MacroTools.postwalk(x -> isexpr(x, :(::), :kw) ? x.args[1] : x, x)
 isvararg(x) = isexpr(x, :(::)) && namify(x.args[2]) == :Vararg
 
+nothings2zeros(x) = x
+nothings2zeros(::Nothing) = Zero()
+nothings2zeros(t::Tuple) = map(nothings2zeros, t)
+
+"""
+    wrap(back)
+
+Wraps the back function in a `nothings2zeros` function, which replaces the `nothing`s with
+`Zero()`s.
+"""
+function wrap(back)
+    function wrapped_back(Δ)
+        return nothings2zeros(back(Δ))
```

Comment:
```suggestion
        return nothings2zeros(back(zeros2nothings(Δ)))
```

Reply:
FYI, I moved this directly in the `_pullback`.

Having 
`return nothings2zeros(back(Δ))`
Gives no errors or warnings.

Whereas
`return nothings2zeros(back(zeros2nothings(Δ)))`
Gives loads of `ZBack` warnings about being fed `nothing`.

How can this happen? It seems to me that `_back` can never be a `chain_rrule`, right? As far as I understand `_back` is the function defined next to `@adjoint`.